platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/net/ConnectivityReceiver.java

```diff
@@ -0,0 +1,74 @@
+package com.mapbox.mapboxsdk.net;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.util.Log;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Listens for Connectivity changes
+ */
+public class ConnectivityReceiver extends BroadcastReceiver {
+    private static final String TAG = ConnectivityReceiver.class.getSimpleName();
+    private static ConnectivityReceiver INSTANCE;
+
+    public static synchronized ConnectivityReceiver instance(Context context) {
+        if (INSTANCE == null) {
+            //Register new instance
+            INSTANCE = new ConnectivityReceiver(context);
+            context.registerReceiver(INSTANCE, new IntentFilter("android.net.conn.CONNECTIVITY_CHANGE"));
+
```

Comment:
No. I think this should remain active to facilitate background activity. We can unregister when/if MapboxAccountManager is released if needed. Why?